autopilot/contractor/contractor_test.go

```diff
@@ -37,62 +39,112 @@ func TestCalculateMinScore(t *testing.T) {
+				storeCost := settings.Prices.RPCAppendSectorsCost(sectors, 1).RenterCost()
+				expectedAllowance = storeCost.Add(uploadCost).Add(downloadCost)
+				expectedCollateral = rhpv4.MaxHostCollateral(settings.Prices, storeCost)
+				return
+			},
 		},
 		{
-			name:                 "UsedAboveMin",
-			minRenterFunds:       50,
-			initRenterFunds:      100,
-			remainingRenterFunds: 90,
-			expected:             90,
+			initialDataSize: 3 << 40, // 3 TiB
+			calc: func(settings rhpv4.HostSettings) (expectedAllowance types.Currency, expectedCollateral types.Currency) {
+				const sectors = maxContractGrowthRate / rhpv4.SectorSize // clamped to 512 GiB
+				uploadCost := settings.Prices.RPCWriteSectorCost(rhpv4.SectorSize).RenterCost().Mul64(sectors)
+				downloadCost := settings.Prices.RPCReadSectorCost(rhpv4.SectorSize).RenterCost().Mul64(sectors)
+				storeCost := settings.Prices.RPCAppendSectorsCost(sectors, 1).RenterCost()
+				expectedAllowance = storeCost.Add(uploadCost).Add(downloadCost)
+				expectedCollateral = rhpv4.MaxHostCollateral(settings.Prices, storeCost)
+				return
+			},
 		},
 		{
-			name:                 "UsedUnderThreshold",
-			minRenterFunds:       0,
-			initRenterFunds:      100,
-			remainingRenterFunds: 70,
-			expected:             80,
+			initialDataSize: 3 << 40, // 3 TiB
+			modSettings: func(settings *rhpv4.HostSettings) {
+				settings.MaxCollateral = types.NewCurrency64(10) // want to test that collateral is clamped to the host's max
+			},
+			calc: func(settings rhpv4.HostSettings) (expectedAllowance types.Currency, expectedCollateral types.Currency) {
+				const sectors = maxContractGrowthRate / rhpv4.SectorSize // clamped to 512 GiB
+				uploadCost := settings.Prices.RPCWriteSectorCost(rhpv4.SectorSize).RenterCost().Mul64(sectors)
+				downloadCost := settings.Prices.RPCReadSectorCost(rhpv4.SectorSize).RenterCost().Mul64(sectors)
+				storeCost := settings.Prices.RPCAppendSectorsCost(sectors, 1).RenterCost()
+				expectedAllowance = storeCost.Add(uploadCost).Add(downloadCost)
+				expectedCollateral = types.NewCurrency64(10) // clamped to the host's max collateral)
+				return
+			},
 		},
 	}
+
 	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			renterFunds := renewFundingEstimate(types.NewCurrency64(test.minRenterFunds), types.NewCurrency64(test.initRenterFunds), types.NewCurrency64(test.remainingRenterFunds), zap.NewNop().Sugar())
-			if !renterFunds.Equals(types.NewCurrency64(test.expected)) {
-				t.Errorf("expected %v but got %v", test.expected, renterFunds)
+		t.Run(fmt.Sprintf("data size %d", test.initialDataSize), func(t *testing.T) {
+			settings := defaultSettings
+			if test.modSettings != nil {
+				test.modSettings(&settings)
+			}
+			expectedAllowance, expectedCollateral := test.calc(settings)
+			allowance, collateral := contractFunding(settings, test.initialDataSize, types.ZeroCurrency, types.ZeroCurrency, 1)
```

Comment:
Good test but it might be nice to also test the min values for allowance and collateral instead of always passing `types.ZeroCurrency`.